qi-lib/flow/core/deforest-syntax.rkt

```diff
@@ -0,0 +1,293 @@
+  (pattern (~or (#%blanket-template
+                 ((#%host-expression map)
+                  (#%host-expression f)
+                  __))
+                (#%fine-template
+                 ((#%host-expression map)
+                  (#%host-expression f)
+                  _)))))
+
+(define-syntax-class fst-filter-map
+  #:attributes (f)
+  #:datum-literals (#%host-expression #%blanket-template __ _ #%fine-template filter-map)
+  (pattern (~or (#%blanket-template
+                 ((#%host-expression filter-map)
+                  (#%host-expression f)
+                  __))
+                (#%fine-template
+                 ((#%host-expression filter-map)
+                  (#%host-expression f)
+                  _)))))
+
+(define-syntax-class fst-take
+  #:attributes (n)
+  #:datum-literals (#%host-expression #%blanket-template __ _ #%fine-template take)
+  (pattern (~or (#%blanket-template
+                 ((#%host-expression take)
+                  __
+                  (#%host-expression n)))
+                (#%fine-template
+                 ((#%host-expression take)
+                  _
+                  (#%host-expression n))))))
+
+(define-syntax-class fusable-stream-transformer0
+  (pattern (~or filter:fst-filter
+                filter-map:fst-filter-map)))
+
+(define-syntax-class fusable-stream-transformer
+  (pattern (~or filter:fst-filter
+                map:fst-map
+                filter-map:fst-filter-map
+                take:fst-take)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Consumers
+
+(define-syntax-class fsc-foldr
+  #:attributes (op init)
+  #:datum-literals (#%host-expression #%blanket-template __ _ #%fine-template foldr)
+  (pattern (~or (#%blanket-template
+                 ((#%host-expression foldr)
+                  (#%host-expression op)
+                  (#%host-expression init)
+                  __))
+                (#%fine-template
+                 ((#%host-expression foldr)
+                  (#%host-expression op)
+                  (#%host-expression init)
+                  _)))))
+
+(define-syntax-class fsc-foldl
+  #:attributes (op init)
+  #:datum-literals (#%host-expression #%blanket-template __ _ #%fine-template foldl)
+  (pattern (~or (#%blanket-template
+                 ((#%host-expression foldl)
+                  (#%host-expression op)
+                  (#%host-expression init)
+                  __))
+                (#%fine-template
+                 ((#%host-expression foldl)
+                  (#%host-expression op)
+                  (#%host-expression init)
+                  _)))))
+
+(define-syntax-class cad*r-datum
+  #:attributes (countdown)
+  (pattern (~datum car) #:attr countdown #'0)
+  (pattern (~datum cadr) #:attr countdown #'1)
+  (pattern (~datum caddr) #:attr countdown #'2)
+  (pattern (~datum cadddr) #:attr countdown #'3)
+  (pattern (~datum caddddr) #:attr countdown #'4)
+  (pattern (~datum cadddddr) #:attr countdown #'5))
```

Comment:
This only adds to my confusion about this commit, since this is the same code as
in 488fd48 (Deforest cadr, caddr, cadddr, caddddr, cadddddr., 2024-03-11).